wire/msgtx.go

```diff
@@ -338,6 +432,22 @@ func (msg *MsgTx) BtcEncode(w io.Writer, pver uint32) error {
 		return err
 	}
 
+	// If the pver is set to WitnessVersion, and the Flags field for the
+	// MsgTx aren't 0x00, then this indicates the transaction is to be
+	// encoded using the new witness inclusionary structure defined in
+	// BIP0141.
+	if pver == WitnessVersion {
```

Comment:
Note that this is an out dated diff. Looks I goofed a bit re-organizing the commits. I now utilize a separate `wire.MessageEncoding` field to gate this behavior.